src/client/dfs/dfs.c

```diff
@@ -1424,34 +1647,49 @@ open_sb(daos_handle_t coh, bool create, daos_obj_id_t super_oid,
 	dfs_layout_ver_t	layout_ver;
 	daos_size_t		chunk_size = 0;
 	daos_oclass_id_t	oclass = OC_UNKNOWN;
+	daos_oclass_id_t	dir_oclass = OC_UNKNOWN;
 	uint32_t		mode;
+	char			hints[DAOS_CONT_HINT_MAX_LEN];
 	int			i, rc;
 
+	D_ASSERT(attr);
+
 	/** Open SB object */
-	rc = daos_obj_open(coh, super_oid, create ? DAOS_OO_RW : DAOS_OO_RO,
-			   oh, NULL);
+	rc = daos_obj_open(coh, super_oid, create ? DAOS_OO_RW : DAOS_OO_RO, oh, NULL);
 	if (rc) {
 		D_ERROR("daos_obj_open() Failed, "DF_RC"\n", DP_RC(rc));
 		return daos_der2errno(rc);
 	}
 
-	d_iov_set(&sg_iovs[0], &magic, sizeof(dfs_magic_t));
-	d_iov_set(&sg_iovs[1], &sb_ver, sizeof(dfs_sb_ver_t));
-	d_iov_set(&sg_iovs[2], &layout_ver, sizeof(dfs_layout_ver_t));
-	d_iov_set(&sg_iovs[3], &chunk_size, sizeof(daos_size_t));
-	d_iov_set(&sg_iovs[4], &oclass, sizeof(daos_oclass_id_t));
-	d_iov_set(&sg_iovs[5], &mode, sizeof(uint32_t));
+	d_iov_set(&sg_iovs[MAGIC_IDX], &magic, sizeof(dfs_magic_t));
+	d_iov_set(&sg_iovs[SB_VER_IDX], &sb_ver, sizeof(dfs_sb_ver_t));
+	d_iov_set(&sg_iovs[LO_VER_IDX], &layout_ver, sizeof(dfs_layout_ver_t));
+	d_iov_set(&sg_iovs[CS_IDX], &chunk_size, sizeof(daos_size_t));
+	d_iov_set(&sg_iovs[FILE_OC_IDX], &oclass, sizeof(daos_oclass_id_t));
+	d_iov_set(&sg_iovs[DIR_OC_IDX], &dir_oclass, sizeof(daos_oclass_id_t));
+	d_iov_set(&sg_iovs[CONT_MODE_IDX], &mode, sizeof(uint32_t));
 
 	for (i = 0; i < SB_AKEYS; i++) {
 		sgls[i].sg_nr		= 1;
 		sgls[i].sg_nr_out	= 0;
 		sgls[i].sg_iovs		= &sg_iovs[i];
 	}
 
-	set_inode_params(create, iods, &dkey);
+	set_sb_params(create, iods, &dkey);
 
 	/** create the SB and exit */
 	if (create) {
+		int num_iods = SB_AKEYS;
+		size_t hint_len = strlen(attr->da_hints);
```

Comment:
Should we use `strnlen` for added robustness? I remember there was some security/coverity/etc. concern about these functions

Reply:
at this point we have already guarded against a larger buffer before this call, so this is safe here.